compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/error.object-shape-value-of.js

```diff
@@ -0,0 +1,18 @@
+import {ValidateMemoization} from 'shared-runtime';
+
+function Component() {
+  const x = {
+    valueOf() {
+      return this;
```

Comment:
My bad, can you try changing this to `return x`? 

Another option would be:

```js
const x ={};
const y = {
  x,
  valueOf() {
    return x;
  }
}
y.valueOf().z = true; // this modifies the “x” object
return <ValudateMemo …
```

Reply:
I added the second one. Thanks!